distributed/cli/tests/test_dask_worker.py

```diff
@@ -280,6 +282,17 @@ def test_nprocs_expands_name(loop):
                     assert len(set(names)) == 4
 
 
+def test_nprocs_deprecation():
+
+    runner = CliRunner()
+    with pytest.warns(UserWarning, match="nprocs"):
+        try:
+            runner.invoke(distributed.cli.dask_worker.main, ["--nprocs=2"])
+        except ValueError:
+            # didn't pass scheduler
+            pass
```

Comment:
As is, this test checks that a warning is emitted when `--nprocs` is specified, but doesn't check that the CLI option still works. We could instead make this test more similar to `test_num_workers_negative` where we do spin up a scheduler and client and check that when, for example, `--nprocs 2` is passed to `dask-worker` that we not only get the expected deprecation warning, but two workers are also created